pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -643,17 +688,17 @@ private InstancePartitions getInstancePartitionsForTier(TableConfig tableConfig,
         LOGGER.info("Persisting instance partitions: {} to ZK", instancePartitions);
         InstancePartitionsUtils.persistInstancePartitions(_helixManager.getHelixPropertyStore(), instancePartitions);
       }
-      return instancePartitions;
+      return Pair.of(instancePartitions, instancePartitions.equals(existingInstancePartitions));
```

Comment:
We want to follow the same logic as non-tier instance assignment. Also, in `bootstrap` mode, we still want to read the `existingInstancePartitions`, but just passing `null` into `instanceAssignmentDriver.assignInstances()`